tests/playwright/src/specs/extension-installation-smoke.spec.ts

```diff
@@ -140,6 +141,26 @@ for (const { extensionName, extensionType } of extentionTypes) {
         await playExpect(extensionPage.status).toHaveText(ACTIVE);
       });
 
+      test('Extension has no errors present', async () => {
+        const extensionsPage = await navigationBar.openExtensions();
+        const extensionPage = await extensionsPage.openExtensionDetails(extensionName, extensionLabel, extensionType);
+        // tabs are empty in case there is no error. If there is error, there are two tabs' buttons present
+        const errorTab = extensionPage.tabs.getByRole('button', { name: 'Error' });
+        // we would like to propagate the error's stack trace into test failure message
+        let stackTrace = '';
+        await waitUntil(
+          async () => {
+            if ((await errorTab.count()) > 0) {
+              stackTrace = await errorTab.innerText();
+              return true;
+            }
+            return false;
+          },
+          { timeout: 3500, sendError: false },
+        );
```

Comment:
I don't really understand why this is needed, can't we just check for the error tab count when we open the page and that's it? Why is this logic needed? Is there some sort of innate delay that makes the error tabs show up later than the page opening? And if so how do you know 3.5 seconds is the correct amount of time?

Reply:
This is for the stacktrace message. Check for errorTab in expect happens afterwards. but if there will be error, we wont know what it was, just locator expect assertion error. but no info about actual error. Nothing than this.

Reply:
No, that part I understand, what I don't get is why does it need to be in an waitUntil() method? what is wrong with the following code:

```suggestion
            if ((await errorTab.count()) > 0) {
              stackTrace = await errorTab.innerText();
            }
```

Reply:
Count does not wait, it evaluates immediately, right? So it seems safer, we do multiple call of the count this way.

Reply:
Which was exactly my question, why do we need to make multiple calls to count() ? Is there some practical reason this is required? And if so how do you know that 3.5 seconds worth of count() calls is the correct amount? Why not 1 second? Why not 5 seconds?

Reply:
why not 3.5? seems like compromise between 1 and 5. Not long, not short. :)
Waits are good in UI testing. There is a lot situations where test might be faster than UI which is being rendered. I am claiming this is good practice. Question is whether we get into state where error tab is not yet present when we reach this code. I will remove the wait and we see if we got empty stackTrace in the message if there will be error.

Reply:
Because 3.5 is too long is a wait isn't actually required and might be too short if it is. But more to the point my impression is that the page is generated with the error tabs when you enter the extension details page so I don't see why any wait would be required **as long as the navigation to the extension page is finished** which I'm guessing could be ensured with `await playExpect(relevant locator).toBeVisible()` where the navigation is being done.

I don't see any navigation in the code snippet that is being changed so I assume the navigation is being done in osme other place.

Reply:
It is done inside `openExtensionDetails` method. So if that is correct, then it is alright. I just do not know how that works inside.

Reply:
in that case I would say that adding `awayt playExpect(extensionPage.heading).toBeVisible();` on line 146 of the spec file is good practice, this ensure the navigation is finished.

Reply:
in doing.